packages/kbn-proc-runner/observe_child_process.js

```diff
@@ -0,0 +1,38 @@
+import Rx from 'rxjs/Rx';
+
+import { createCliError } from './errors';
+
+/**
+ *  Creates an Observable from a childProcess that:
+ *    - provides the exit code as it's only value (which may be null)
+ *      as soon as the process exits
+ *    - completes once all stdio streams for the child process have closed
+ *    - fails if the childProcess emits an error event
+ *
+ *  @param  {ChildProcess} childProcess
+ *  @return {Rx.Observable}
+ */
+export function observeChildProcess(name, childProcess) {
+  // observe first exit event
+  const exit$ = Rx.Observable.fromEvent(childProcess, 'exit')
+    .first()
+    .map(code => {
+      // JVM exits with 143 on SIGTERM and 130 on SIGINT, dont' treat then as errors
+      if (code > 0 && !(code === 143 || code === 130)) {
+        throw createCliError(`[${name}] exitted with code ${code}`);
+      }
+
+      return code;
+    });
+
+  // observe first close event
+  const close$ = Rx.Observable.fromEvent(childProcess, 'close').first();
```

Comment:
I've noticed errors from using `.first()` in the original version of this code, caused by the fact that sometimes there isn't a `.close()` event, mind switching this and the ones above/below to `.take(1)`?